This fails to compile with:

```
error[E0407]: method `new_frame_ready` is not a member of trait `webrender_api::RenderNotifier`
   --> components/compositing/compositor.rs:332:5
    |
332 | /     fn new_frame_ready(&self) {
333 | |         self.compositor_proxy.recomposite(CompositingReason::NewWebRenderFrame);
334 | |     }
    | |_____^ not a member of trait `webrender_api::RenderNotifier`

error[E0407]: method `new_scroll_frame_ready` is not a member of trait `webrender_api::RenderNotifier`
   --> components/compositing/compositor.rs:336:5
    |
336 | /     fn new_scroll_frame_ready(&self, composite_needed: bool) {
337 | |         self.compositor_proxy.send(Msg::NewScrollFrameReady(composite_needed));
338 | |     }
    | |_____^ not a member of trait `webrender_api::RenderNotifier`

error[E0046]: not all trait items implemented, missing: `wake_up`, `new_document_ready`
   --> components/compositing/compositor.rs:327:1
    |
327 | / impl webrender_api::RenderNotifier for RenderNotifier {
328 | |     fn clone(&self) -> Box<webrender_api::RenderNotifier> {
329 | |         Box::new(RenderNotifier::new(self.compositor_proxy.clone()))
330 | |     }
...   |
338 | |     }
339 | | }
    | |_^ missing `wake_up`, `new_document_ready` in implementation
    |
    = note: `wake_up` from trait: `fn(&Self)`
    = note: `new_document_ready` from trait: `fn(&Self, webrender_api::DocumentId, bool, bool)`

error: aborting due to 3 previous errors
```

Still, I wanted to push it because I assume @glennw, @pcwalton or whoever will do the code changes will appreciate having the Cargo changes ready anyway.

--- a/components/compositing/compositor.rs
+++ b/components/compositing/compositor.rs
@@ -329,12 +329,21 @@ impl webrender_api::RenderNotifier for RenderNotifier {
         Box::new(RenderNotifier::new(self.compositor_proxy.clone()))
     }
 
-    fn new_frame_ready(&self) {
+    fn wake_up(&self) {
         self.compositor_proxy.recomposite(CompositingReason::NewWebRenderFrame);
     }
 
-    fn new_scroll_frame_ready(&self, composite_needed: bool) {
-        self.compositor_proxy.send(Msg::NewScrollFrameReady(composite_needed));
+    fn new_document_ready(
+        &self,
+        _document_id: webrender_api::DocumentId,
+        scrolled: bool,
+        composite_needed: bool,
+    ) {
+        if scrolled {
+            self.compositor_proxy.send(Msg::NewScrollFrameReady(composite_needed));
+        } else {
+            self.wake_up();
+        }
     }
 }
 

--- a/components/gfx/Cargo.toml
+++ b/components/gfx/Cargo.toml
@@ -48,8 +48,8 @@ xi-unicode = "0.1.0"
 [target.'cfg(target_os = "macos")'.dependencies]
 byteorder = "1.0"
 core-foundation = "0.4"
-core-graphics = "0.9"
-core-text = "7.0"
+core-graphics = "0.12.2"
+core-text = "8.0"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies]
 freetype = "0.3"

--- a/components/gfx/platform/macos/font_template.rs
+++ b/components/gfx/platform/macos/font_template.rs
@@ -17,7 +17,7 @@ use std::fmt;
 use std::fs::File;
 use std::io::{Read, Error as IoError};
 use std::ops::Deref;
-use std::sync::Mutex;
+use std::sync::{Arc, Mutex};
 use webrender_api::NativeFontHandle;
 
 /// Platform specific font representation for mac.
@@ -36,7 +36,7 @@ pub struct FontTemplateData {
     ctfont: CachedCTFont,
 
     pub identifier: Atom,
-    pub font_data: Option<Vec<u8>>
+    pub font_data: Option<Arc<Vec<u8>>>
 }
 
 unsafe impl Send for FontTemplateData {}
@@ -47,7 +47,7 @@ impl FontTemplateData {
         Ok(FontTemplateData {
             ctfont: CachedCTFont(Mutex::new(HashMap::new())),
             identifier: identifier.to_owned(),
-            font_data: font_data
+            font_data: font_data.map(Arc::new)
         })
     }
 
@@ -61,7 +61,7 @@ impl FontTemplateData {
             let clamped_pt_size = pt_size.max(0.01);
             let ctfont = match self.font_data {
                 Some(ref bytes) => {
-                    let fontprov = CGDataProvider::from_buffer(bytes);
+                    let fontprov = CGDataProvider::from_buffer(bytes.clone());
                     let cgfont_result = CGFont::from_data_provider(fontprov);
                     match cgfont_result {
                         Ok(cgfont) => {
@@ -103,7 +103,7 @@ impl FontTemplateData {
     /// Returns a clone of the bytes in this font if they are in memory. This function never
     /// performs disk I/O.
     pub fn bytes_if_in_memory(&self) -> Option<Vec<u8>> {
-        self.font_data.clone()
+        self.font_data.as_ref().map(|bytes| (**bytes).clone())
     }
 
     /// Returns the native font that underlies this font template, if applicable.

--- a/components/servo/lib.rs
+++ b/components/servo/lib.rs
@@ -204,7 +204,8 @@ impl<Window> Servo<Window> where Window: WindowMethods + 'static {
         };
 
         let webrender_api = webrender_api_sender.create_api();
-        let webrender_document = webrender_api.add_document(window.framebuffer_size());
+        let wr_document_layer = 0; //TODO
+        let webrender_document = webrender_api.add_document(window.framebuffer_size(), wr_document_layer);
 
         // Important that this call is done in a single-threaded fashion, we
         // can't defer it after `create_constellation` has started.

--- a/ports/cef/Cargo.toml
+++ b/ports/cef/Cargo.toml
@@ -40,7 +40,7 @@ webrender_api = {git = "https://github.com/servo/webrender", features = ["ipc"]}
 
 [target.'cfg(target_os="macos")'.dependencies]
 objc = "0.2"
-cocoa = "0.10"
+cocoa = "0.13"
 
 [target.'cfg(target_os="linux")'.dependencies]
 x11 = "2.3"

--- a/ports/glutin/Cargo.toml
+++ b/ports/glutin/Cargo.toml
@@ -18,7 +18,7 @@ log = "0.3.5"
 msg = {path = "../../components/msg"}
 net_traits = {path = "../../components/net_traits"}
 script_traits = {path = "../../components/script_traits"}
-servo-glutin = "0.12"
+servo-glutin = "0.13"
 servo_geometry = {path = "../../components/geometry"}
 servo_config = {path = "../../components/config"}
 servo_url = {path = "../../components/url"}

--- a/servo-tidy.toml
+++ b/servo-tidy.toml
@@ -36,6 +36,7 @@ num = []
 packages = [
   "bitflags",
   "cookie",
+  "foreign-types",
   "mime",
   "semver",
   "toml",

